src/util.hpp

```diff
@@ -52,59 +52,49 @@ mapbox::geometry::point<std::int64_t> create_query_point(double lng,
                                                          std::uint32_t extent,
                                                          std::uint32_t active_tile_x,
                                                          std::uint32_t active_tile_y) {
-    std::uint32_t z2 = 1 << zoom; // number of tiles 'across' a particular zoom level
 
-    lng = fmod((lng + 180.0), 360.0);
+    lng = std::fmod((lng + 180.0), 360.0);
     if (lat > 89.9) {
         lat = 89.9;
     } else if (lat < -89.9) {
         lat = -89.9;
     }
 
+    double z2 = static_cast<double>(1 << zoom); // number of tiles 'across' a particular zoom level
     double lat_radian = (lat * M_PI) / 180.0;
     double zl_x = lng / (360.0 / (extent * z2));
-    double zl_y = ((extent * z2) / 2) * (1.0 - (log(tan(lat_radian) + 1.0 / cos(lat_radian)) / M_PI));
-    // std::clog << "zl_x: " << zl_x << ", zl_y: " << zl_y << std::endl;
-
-    std::uint64_t origin_tile_x = static_cast<std::uint64_t>(floor(zl_x / extent));
-    std::uint64_t origin_tile_y = static_cast<std::uint64_t>(floor(zl_y / extent));
-    // std::clog << "origin tile x: " << origin_tile_x << ", origin tile y: " << origin_tile_y << std::endl;
-    // std::clog << "active tile x: " << active_tile_x << ", active tile y: " << active_tile_y << std::endl;
-
-    std::uint64_t origin_x = std::uint32_t(fmod(floor(zl_x), extent));
-    std::uint64_t origin_y = std::uint32_t(fmod(floor(zl_y), extent));
-    // std::clog << "origin tile coordinate x: " << origin_x << ", origin tile coordinate y: " << origin_y << std::endl;
-
-    auto diff_tile_x = active_tile_x - origin_tile_x;
-    auto diff_tile_y = active_tile_y - origin_tile_y;
-    // std::clog << "tile difference x: " << diff_tile_x << ", y: " << diff_tile_y << std::endl;
-    auto query_x = -(diff_tile_x * extent) + origin_x;
-    auto query_y = -(diff_tile_y * extent) + origin_y;
-    // std::clog << "query point x: " << query_x << ", y: " << query_y << std::endl;
-    mapbox::geometry::point<std::int64_t> query_point{static_cast<std::int64_t>(query_x), static_cast<std::int64_t>(query_y)};
+    double zl_y = ((extent * z2) / 2) * (1.0 - (std::log(std::tan(lat_radian) + 1.0 / std::cos(lat_radian)) / M_PI));
+    std::int64_t origin_tile_x = static_cast<std::int64_t>(std::floor(zl_x / extent));
+    std::int64_t origin_tile_y = static_cast<std::int64_t>(std::floor(zl_y / extent));
+    std::int64_t origin_x = std::int64_t(std::fmod(std::floor(zl_x), extent));
```

Comment:
You can use a standard modulus here assuming that you convert zl_x to a std::int64_t first.

Reply:
@flippmoke can't seem to get the types 💯 correct here. Mind pushing a change up so I can see?